diff --git a/node_modules/@types/node-forge/index.d.ts b/node_modules/@types/node-forge/index.d.ts
index b95d0ab..53647fd 100755
--- a/node_modules/@types/node-forge/index.d.ts
+++ b/node_modules/@types/node-forge/index.d.ts
@@ -252,7 +252,7 @@ declare module 'node-forge' {
     namespace pki {
         type PEM = string;
         type PublicKey = rsa.PublicKey | ed25519.Key;
-        type PrivateKey = rsa.PrivateKey | ed25519.Key;
+        type PrivateKey = rsa.PrivateKey | ed25519.Key | ecdsa.PrivateKey;
         type EncryptionOptions = {
             algorithm?: 'aes128' | 'aes192' | 'aes256' | '3des' | undefined;
             count?: number | undefined;
@@ -714,7 +714,14 @@ declare module 'node-forge' {
             options?: FingerprintOptions,
         ): util.ByteStringBuffer | Hex | string;
     }
+    namespace ecdsa{
+        class PrivateKey{
+            constructor(ec, privateKey);
+        }
 
+        function privateKeyToAsn1(privateKey: PrivateKey): asn1.Asn1;
+        function privateKeyFromAsn1(privateKey: asn1.Asn1): PrivateKey;
+    }
     namespace asn1 {
         enum Class {
             UNIVERSAL = 0x00,
