diff --git a/node_modules/@types/node-forge/index.d.ts b/node_modules/@types/node-forge/index.d.ts
index 54ed539..6356825 100755
--- a/node_modules/@types/node-forge/index.d.ts
+++ b/node_modules/@types/node-forge/index.d.ts
@@ -253,7 +253,7 @@ declare module 'node-forge' {
     namespace pki {
         type PEM = string;
         type PublicKey = rsa.PublicKey | ed25519.Key;
-        type PrivateKey = rsa.PrivateKey | ed25519.Key;
+        type PrivateKey = rsa.PrivateKey | ed25519.Key | ecdsa.PrivateKey;
         type EncryptionOptions = {
             algorithm?: 'aes128' | 'aes192' | 'aes256' | '3des' | undefined;
             count?: number | undefined;
@@ -370,12 +370,12 @@ declare module 'node-forge' {
 
             type ToNativeBufferParameters =
                 | {
-                      message: NativeBuffer | util.ByteBuffer;
-                  }
+                    message: NativeBuffer | util.ByteBuffer;
+                }
                 | {
-                      message: string;
-                      encoding: 'binary' | 'utf8';
-                  };
+                    message: string;
+                    encoding: 'binary' | 'utf8';
+                };
 
             // `string`s will be converted by toNativeBuffer with `encoding: 'binary'`
             type BinaryBuffer = NativeBuffer | util.ByteBuffer | string;
@@ -643,11 +643,11 @@ declare module 'node-forge' {
             options?:
                 | ((verified: boolean | string, depth: number, certs: Certificate[]) => boolean)
                 | {
-                      verify?:
-                          | ((verified: boolean | string, depth: number, certs: Certificate[]) => boolean)
-                          | undefined;
-                      validityCheckDate?: Date | null | undefined;
-                  },
+                    verify?:
+                    | ((verified: boolean | string, depth: number, certs: Certificate[]) => boolean)
+                    | undefined;
+                    validityCheckDate?: Date | null | undefined;
+                },
         ): boolean;
 
         function pemToDer(pem: PEM): util.ByteStringBuffer;
@@ -699,7 +699,13 @@ declare module 'node-forge' {
         function getPublicKeyFingerprint(publicKey: PublicKey, options: HexFingerprintOptions): Hex;
         function getPublicKeyFingerprint(publicKey: PublicKey, options: BinaryFingerprintOptions): Bytes;
     }
-
+    namespace ecdsa {
+        class PrivateKey {
+            constructor(ec, privateKey);
+        }
+        function privateKeyToAsn1(privateKey: PrivateKey): asn1.Asn1;
+        function privateKeyFromAsn1(privateKey: asn1.Asn1): PrivateKey;
+    }
     namespace random {
         function getBytes(count: number, callback?: (err: Error | null, bytes: Bytes) => any): void;
         function getBytesSync(count: number): Bytes;
@@ -1382,12 +1388,12 @@ declare module 'node-forge' {
             verifyClient: boolean;
             verify(conn: Connection, verified: Verified, depth: number, certs: pki.Certificate[]): Verified;
             getCertificate:
-                | ((conn: Connection, hint: CertificateRequest | string[]) => pki.PEM | ReadonlyArray<pki.PEM>)
-                | null;
+            | ((conn: Connection, hint: CertificateRequest | string[]) => pki.PEM | ReadonlyArray<pki.PEM>)
+            | null;
             getPrivateKey: ((conn: Connection, certificate: pki.Certificate) => pki.PEM) | null;
             getSignature:
-                | ((conn: Connection, bytes: Bytes, callback: (conn: Connection, bytes: Bytes) => void) => void)
-                | null;
+            | ((conn: Connection, bytes: Bytes, callback: (conn: Connection, bytes: Bytes) => void) => void)
+            | null;
             input: util.ByteBuffer;
             tlsData: util.ByteBuffer;
             data: util.ByteBuffer;
