diff --git a/node_modules/@types/node-forge/index.d.ts b/node_modules/@types/node-forge/index.d.ts
index 4cbd4ac..056a0ea 100755
--- a/node_modules/@types/node-forge/index.d.ts
+++ b/node_modules/@types/node-forge/index.d.ts
@@ -253,7 +253,7 @@ declare module 'node-forge' {
     namespace pki {
         type PEM = string;
         type PublicKey = rsa.PublicKey | ed25519.Key;
-        type PrivateKey = rsa.PrivateKey | ed25519.Key;
+        type PrivateKey = rsa.PrivateKey | ed25519.Key | ecdsa.PrivateKey;
         type EncryptionOptions = {
             algorithm?: 'aes128' | 'aes192' | 'aes256' | '3des' | undefined;
             count?: number | undefined;
@@ -703,6 +703,14 @@ declare module 'node-forge' {
         function getPublicKeyFingerprint(publicKey: PublicKey, options: BinaryFingerprintOptions): Bytes;
     }
 
+    namespace ecdsa {
+        class PrivateKey {
+            constructor(ec, privateKey);
+        }
+        function privateKeyToAsn1(privateKey: PrivateKey): asn1.Asn1;
+        function privateKeyFromAsn1(privateKey: asn1.Asn1): PrivateKey;
+    }
+    
     namespace random {
         function getBytes(count: number, callback?: (err: Error | null, bytes: Bytes) => any): void;
         function getBytesSync(count: number): Bytes;
